### PR TITLE
Backport https://github.com/apache/incubator-beam/pull/1230

### PR DESCRIPTION
**Pull request by sammcveety**
R: @dhalperi 

Pulled in changes for PubsubUnbounded*.java from https://github.com/apache/incubator-beam/pull/1504

### CODE REVIEW

**Comment by sammcveety** [comment]
Rebased

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java [hunk 1]
```diff
@@ -666,8 +754,11 @@ private Bound(String name, PubsubSubscription subscription, PubsubTopic topic,
 
       @Override
       public void populateDisplayData(DisplayData.Builder builder) {
-        super.populateDisplayData(builder);
-        populateCommonDisplayData(builder, timestampLabel, idLabel, topic);
+        String topicString =
+            topic == null ? null
+            : topic.isAccessible() ? topic.get().asPath()
+            : topic.toString();
+        populateCommonDisplayData(builder, timestampLabel, idLabel, topicString);
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java]: Shouldn't we pass the VP into `populateCommonDisplayData` and avoid the logic here?

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/PubsubIOTest.java [hunk 2]
```diff
@@ -70,29 +73,13 @@ public void testTopicValidationSuccess() throws Exception {
   }
 
   @Test
-  public void testTopicValidationBadCharacter() throws Exception {
```
Comment (dhalperi) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/PubsubIOTest.java]: Not sure I like this change. Given an SVP, we should be able to catch these bad names, shouldn't we?

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java (hunk 1 above)
Comment (sammcveety) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java]: Yes, yes we should.

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/PubsubIOTest.java (hunk 2 above)
Comment (sammcveety) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/PubsubIOTest.java]: Added it back, agreed.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java [hunk 3]
```diff
@@ -140,15 +140,17 @@ private static void validatePubsubName(String name) {
...
         .addIfNotNull(DisplayData.item("timestampLabel", timestampLabel)
             .withLabel("Timestamp Label Attribute"))
         .addIfNotNull(DisplayData.item("idLabel", idLabel)
             .withLabel("ID Label Attribute"));
 
     if (topic != null) {
-      builder.add(DisplayData.item("topic", topic)
+      String topicString = topic.isAccessible() ? topic.get().asPath()
+          : topic.toString();
+      builder.add(DisplayData.item("topic", topicString)
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/PubsubIO.java]: Answered: it only accepts VP<KnownType> https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java#L880

**Pull request merged**